modules/widget-visibility/widget-conditions.php

```diff
@@ -45,7 +45,7 @@ public static function widget_admin_setup() {
 
 		$categories = get_categories(
 			array(
-				'number'  => 1000,
+				'number'  => apply_filters( 'jetpack_widget_visibility_max_number_categories', 1000 ),
```

Comment:
Now that we have a filter, it may be good to ensure the data returned by the filter is always an integer, to avoid any errors if someone sends in wrong data.

Reply:
Done in 6239d9e16343d99e08b8c3fc316a52e04735087e